extensions/json-schema/services/src/main/java/org/apache/unomi/schema/api/ValidationError.java

```diff
@@ -42,7 +42,8 @@ public String toString() {
     }
 
     public boolean equals(Object o) {
-        return validationMessage.equals(o);
+        ValidationError other = (ValidationError) o;
+        return validationMessage.equals(other.validationMessage);
```

Comment:
Could be simpler to replace directly the inner ValidationMessage object by the String error message at this step.
I would also simplify the building of custom error messages.